### ISSUE TITLE
netbsd.compat cannot be cross-compiled

### ISSUE THREAD

**Issue by r-burns** [NixOS/nixpkgs#135332]
### Describe the bug
`netbsd.compat` cross-compilation is broken. The package appears to cross-compile successfully, but produces native object files for the build system's architecture rather than the host system. This causes downstream failures when cross-compiling libs which depend on libnbcompat.

### Steps To Reproduce

On an x86_64 builder:
```
> nix-build . -A pkgsCross.muslpi.netbsd.compat
> readelf -h result/lib/libnbcompat.a
<snip>
File: result/lib/libnbcompat.a(rec_utils.lo)
ELF Header:
  Magic:   7f 45 4c 46 02 01 01 00 00 00 00 00 00 00 00 00 
  Class:                             ELF64
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              REL (Relocatable file)
  Machine:                           Advanced Micro Devices X86-64
  Version:                           0x1
  Entry point address:               0x0
  Start of program headers:          0 (bytes into file)
  Start of section headers:          1112 (bytes into file)
  Flags:                             0x0
  Size of this header:               64 (bytes)
  Size of program headers:           0 (bytes)
  Number of program headers:         0
  Size of section headers:           64 (bytes)
  Number of section headers:         12
  Section header string table index: 11
```

### Expected behavior
libnbcompat should produce libraries for the host platform

### Additional context
netbsd.compat is a dependency of netbsd.fts, which is used as a compatibility shim when compiling for musl platforms, because musl libc does not have builtin fts support.

I plan to switch our toplevel fts to use musl-fts until this is resolved. A number of other musl-focused distros use musl-fts for this purpose: https://repology.org/project/musl-fts/versions.

### Notify maintainers
@Ericson2314 

### Metadata

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: netbsd.compat
# a list of nixos modules affected by the problem
module:
```

**Comment by sternenseemann** [comment]
cc @alyssais

**Comment by alyssais** [comment]
This is really tricky, because the Makefile supports cross-compilation, but assumes that you'll always want libnbcompat for the build system (host system in NetBSD terminology), because it's intended to be used for building other build tools that don't get installed.

Convincing it to build for the host system instead is not going to be easy.

**Comment by Ericson2314** [comment]
If you remove all `HOST_` from `tools/compat/Makefile` I think that might actually get you decently close?

**Comment by alyssais** [comment]
I tried that but couldn't get it to work.  Want to have a go?

**Comment by Ericson2314** [comment]
Sure.

**Comment by r-burns** [comment]
Fixed by #136437
